packages/vertexai/src/methods/generate-content.ts

```diff
@@ -58,9 +66,26 @@ export async function generateContent(
     JSON.stringify(params),
     requestOptions
   );
-  const responseJson: GenerateContentResponse = await response.json();
-  const enhancedResponse = createEnhancedContentResponse(responseJson);
+  const generateContentResponse = await handleGenerateContentResponse(
+    response,
+    apiSettings
+  );
+  const enhancedResponse = createEnhancedContentResponse(
+    generateContentResponse
+  );
   return {
     response: enhancedResponse
   };
 }
+
+async function handleGenerateContentResponse(
```

Comment:
`processGenerateContentResponse()`? I think `handle` is conventional for event and error handlers.